src/utils/configurations/configurationsContext.js

```diff
@@ -0,0 +1,231 @@
+    configurations: {
+        resource: DATASTORE_ENDPOINT,
+    },
+}
+
+const LoadingSpinner = () => (
+    <Layer>
+        <CenteredContent>
+            <CircularLoader />
+        </CenteredContent>
+    </Layer>
+)
+
+const ErrorInfo = ({ errorMessage }) => (
+    <Box height={'100%'}>
+        <CenteredContent>
+            <NoticeBox
+                error
+                title={'Error when fetching report configurations'}
+            >
+                {errorMessage}
+            </NoticeBox>
+        </CenteredContent>
+    </Box>
+)
+ErrorInfo.propTypes = { errorMessage: PropTypes.string }
+
+// These contexts are separated so that 'update' consumers don't
+// necessarily rerender when the `configurations` object updates
+// and vice versa (it's a small optimization)
+const ConfigurationsContext = React.createContext()
+const SetConfigurationsContext = React.createContext()
+
+/**
+ * This will fetch configurations once, then maintain it in state.
+ * Further updates should be reflected locally without needing to refetch
+ */
+export const ConfigurationsProvider = ({ children }) => {
+    const [configurations, setConfigurations] = useState({})
+    const { loading, error, data } = useDataQuery(CONFIGURATIONS_QUERY, {
+        onComplete: (data) => {
+            setConfigurations(data.configurations)
+        },
+        // todo: on error, check if the datastore key doesn't exist yet.
+        // if not, create one
+    })
+
+    if (loading) {
+        return <LoadingSpinner />
+    }
+
+    if (error) {
+        console.error(error)
+        return <ErrorInfo errorMessage={error.message} />
+    }
+
+    // this check helps avoid crashes during hot-reloads in development
+    if (!data) {
+        return null
+    }
+
+    return (
+        <ConfigurationsContext.Provider value={configurations}>
+            <SetConfigurationsContext.Provider value={setConfigurations}>
+                {children}
+            </SetConfigurationsContext.Provider>
+        </ConfigurationsContext.Provider>
+    )
+}
+ConfigurationsProvider.propTypes = {
+    children: PropTypes.node,
+}
+
+/** Returns just the configuration state */
+export const useConfigurations = () => {
+    const configurations = useContext(ConfigurationsContext)
+
+    if (!configurations) {
+        throw new Error(
+            'useConfigurations must be used inside of a ConfigurationsProvider'
+        )
+    }
+
+    // return just `configurations`, not `setConfigurations`
+    return configurations
+}
+
+const UPDATE_CONFIGURATIONS_MUTATION = {
+    resource: DATASTORE_ENDPOINT,
+    type: 'update',
+    data: ({ newConfigurations }) => newConfigurations,
+}
+
+/**
+ * Returns updateConfigurations, which accepts a complete new configuration object,
+ * updates the `configurations` state locally, and syncs the new configurations
+ * with the server. It also handles errors with the network request
+ */
+export const useUpdateConfigurations = () => {
```

Comment:
I wouldn't keep this in here as it's not used, and then I think it will potentially confuse people who don't have a background as to why it's here / lead people to potentially use it when they should instead try to use `useConfigurationsDispatch`